tests/common/osutil/test_default.py

```diff
@@ -190,8 +190,11 @@ def test_get_first_if_all_loopback(self):
                 self.assertEqual(('', ''), osutil.DefaultOSUtil().get_first_if())
 
     def test_isloopback(self):
-        self.assertTrue(osutil.DefaultOSUtil().is_loopback('lo'))
-        self.assertFalse(osutil.DefaultOSUtil().is_loopback('eth0'))
+        for iface in osutil.DefaultOSUtil()._get_all_interfaces():
+            if iface == 'lo':
+                self.assertTrue(osutil.DefaultOSUtil().is_loopback(iface))
+            else:
+                self.assertFalse(osutil.DefaultOSUtil().is_loopback(iface))
```

Comment:
Test code should count the number of assertTrue and assertFalse that are actually executed at runtime. If the assertTrue is never fired, the test code should force a failure, because that indicates the testing environment itself is flawed. (If the system-under-test lacks a loopback interface, then the test_isloopback test didn't actually test anything, and the developer needs to know that.)

When unit tests are run under DCR, this is the best way to find a flawed VM added to the DCR fleet.

Reply:
Yes, thanks. That sounds like a good test for _get_all_interfaces. I can add one asserting that it returns exactly 1 loopback and 1 or more non-loopback interfaces. Sounds good?

Reply:
SGTM, thanks.

Reply:
OK, it's added now.